piksi_tools/fileio.py

```diff
@@ -346,7 +346,7 @@ def write(self, filename, data, offset=0, trunc=True, progress_cb=None):
                     data='')
                 sr.send(msg)
                 offset += len(chunk)
-                if progress_cb is not None:
+                if progress_cb is not None and seq % PROGRESS_CB_REDUCTION_FACTOR == 0:
```

Comment:
We could possibly reduce even further if we dynamically calculate `PROGRESS_CB_REDUCTION_FACTOR ` to be 1% of the file size and then even avoid the last cycle where the progress bar would likely 'finish' without actually reaching 100%

Reply:
I'll add that to my master work on upgrade.